### ISSUE TITLE
raft topology: documentation

### ISSUE THREAD

**Issue by kbr-scylla** [scylladb/scylladb#15934]
The new topology-changes-on-raft will have to be documented before going out of experimental: how it's different from the existing topology change procedure from the point of view of the user, and known issues with them (if any).

**Comment by patjed41** [comment]
One thing we must remember is documenting the operator's responsibility for shutting the node down in case when the topology coordinator fails to deliver the join rejection. This change is from #15865.

**Comment by kbr-scylla** [comment]
cc @annastuchlik

**Comment by kbr-scylla** [comment]
Note: we plan to support concurrent bootstrap as well, so we'll have to remember about documenting that as well

**Comment by kbr-scylla** [comment]
Ref: https://github.com/scylladb/scylladb/issues/15735

**Comment by kbr-scylla** [comment]
We'll need to document changes to `removenode` and `replace` after Gleb's `change the way ignored nodes are handled by the topology coordinator` series (a WARNing that removed and ignored nodes are permanently banned even if the request fails)

**Comment by piodul** [comment]
https://github.com/scylladb/scylladb/pull/17077#discussion_r1481112031

Another important piece of information: before starting upgrade to raft topology, the administrator needs to make sure that the `SUPPORTS_CONSISTENT_TOPOLOGY_CHANGES` cluster is enabled on all nodes in the cluster.

**Comment by kbr-scylla** [comment]
Created https://github.com/scylladb/scylladb/issues/17341 which needs to be done by dev team

**Comment by kbr-scylla** [comment]
We need to document that resharding in raft-topology mode requires quorum of nodes to be alive

@vladzcloudius heads up -- you asked to communicate such things

**Comment by kbr-scylla** [comment]
You mean like a placeholder page to be filled later?

**Comment by kbr-scylla** [comment]
> We're already using host ID. https://opensource.docs.scylladb.com/master/operating-scylla/nodetool-commands/removenode.html and there is a warning about deprecating the IP use.

It says:

> However, this feature is deprecated with consistent topology updates enabled

Can we remove the conditional? The feature is deprecated, period.
I see we have this note in 5.4 too, where "consistent topology updates" is still experimental, so definitely would remove it there.

**Comment by kbr-scylla** [comment]
IIUC all required things were done in 6f374aa7d68e297b2f69df208a41eedf15507a15.
The upgrade/recovery procedure has a separate issue: https://github.com/scylladb/scylladb/issues/17341

so we can close this one.
